nixos/modules/services/monitoring/ups.nix

```diff
@@ -444,51 +441,51 @@ in
         '';
       };
 
-      upsmon = mkOption {
+      upsmon = lib.mkOption {
         default = {};
         description = ''
           Options for the `upsmon.conf` configuration file.
         '';
-        type = types.submodule upsmonOptions;
+        type = lib.types.submodule upsmonOptions;
       };
 
-      upsd = mkOption {
+      upsd = lib.mkOption {
         default = {};
         description = ''
           Options for the `upsd.conf` configuration file.
         '';
-        type = types.submodule upsdOptions;
+        type = lib.types.submodule upsdOptions;
       };
 
-      ups = mkOption {
+      ups = lib.mkOption {
         default = {};
         # see nut/etc/ups.conf.sample
         description = ''
           This is where you configure all the UPSes that this system will be
           monitoring directly.  These are usually attached to serial ports,
           but USB devices are also supported.
         '';
-        type = with types; attrsOf (submodule upsOptions);
+        type = with lib.types; attrsOf (submodule upsOptions);
       };
 
-      users = mkOption {
+      users = lib.mkOption {
         default = {};
         description = ''
           Users that can access upsd. See `man upsd.users`.
         '';
-        type = with types; attrsOf (submodule userOptions);
+        type = with lib.types; attrsOf (submodule userOptions);
       };
 
     };
   };
 
-  config = mkIf cfg.enable {
+  config = lib.mkIf cfg.enable {
 
     assertions = [
       (let
-        totalPowerValue = foldl' add 0 (map (monitor: monitor.powerValue) (attrValues cfg.upsmon.monitor));
+        totalPowerValue = lib.foldl' lib.add 0 (map (monitor: monitor.powerValue) (lib.attrValues cfg.upsmon.monitor));
```

Comment:
`lib.add` 👀

Reply:
@philiptaron should we better use `add`/`builtins.add`/`lib.add`

https://github.com/NixOS/nixpkgs/blob/master/lib/trivial.nix#L354

Reply:
It's OK. I have at this point memorized most of the commonly used identifiers that are in `lib`, so that when I'm scanning through the PR, ones that I don't see often jump out at me. This one made me search above and below for an `add` token.